### PR TITLE
move to pyproject.toml

### PR DESCRIPTION
**Pull request by cmacdonald**
closes: #489 

 - [x] how to override version for test releases to pypi
 - [ ] update wiki on how to do a release

### CODE REVIEW

**Comment by cmacdonald** [comment]
> Perhaps sed `__init__.py` to modify the version in the source?

Thanks for the hint. This was somehow easier than trying to rewrite a the toml file, or find a way to override it.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
+  "Operating System :: OS Independent",
+]
+dynamic = ["version", "dependencies"]
+
+[tool.setuptools.dynamic]
+version = {attr = "pyterrier.__version__"}
+dependencies = {file = ["requirements.txt"]}
+
+#package_data={'': ['LICENSE.txt', 'requirements.txt', 'requirements-test.txt']}
```
Comment (cmacdonald) [inline pyproject.toml:29]: are these included?

Comment (cmacdonald) [reply pyproject.toml:29]: do we need requirements.txt included?

Comment (seanmacavaney) [reply pyproject.toml:29]: they don't need to be included

**Review by seanmacavaney — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (cmacdonald) [reply pyproject.toml:29]: you're the expert on this new format - can you make the changes?

**Pull request merged**